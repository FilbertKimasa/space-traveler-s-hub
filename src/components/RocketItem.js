import React, { Dispatch } from 'react';
import PropTypes from 'prop-types';
import { bookRocket } from '../redux/rockets/rocketsSlice';
import '../styles/RocketItem.css';

const dispatch = Dispatch();
function RocketItem({ itemProp }) {
  return (
    <>
      <li className="rocket-item">
        <img
          src={itemProp.flickr_images[0]}
          alt="rocket"
          className="rocket-img"
        />
        <span className="display-flex rocket-info">
          <h2>{itemProp.name}</h2>
          <p>{itemProp.description}</p>
          <button
            className="reserve-btn"
            type="button"
            onClick={() => dispatch(bookRocket({ id: itemProp.id }))}
          >
            Reserve Rocket
          </button>
        </span>
      </li>
    </>
  );
}

RocketItem.propTypes = {
  itemProp: PropTypes.objectOf(
    PropTypes.shape({
      flickr_images: PropTypes.arrayOf(PropTypes.string).isRequired,
      name: PropTypes.string.isRequired,
      itemProp: PropTypes.string.isRequired,
    }),
  ).isRequired,
};

export default RocketItem;
