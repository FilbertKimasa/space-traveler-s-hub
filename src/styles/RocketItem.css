.rocket-img {
  width: 100px;
  height: auto;
}
