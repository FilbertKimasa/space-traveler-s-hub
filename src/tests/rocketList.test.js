import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import RocketItem from '../components/RocketItem';

const mockStore = configureStore([]);

describe('RocketItem component', () => {
  test('renders rocket item correctly', () => {
    const store = mockStore({});

    const rocket = {
      flickr_images: ['rocket-image.jpg'],
      id: 1,
      name: 'Rocket 1',
      description: 'This is a rocket.',
      reserved: false,
    };

    const { container } = render(
      <Provider store={store}>
        <RocketItem itemProp={rocket} />
      </Provider>,
    );

    expect(container.innerHTML).toMatchInlineSnapshot('"<li class=\\"rocket-item\\"><img src=\\"rocket-image.jpg\\" alt=\\"rocket\\" class=\\"rocket-img\\"><span class=\\"display-flex rocket-info\\"><h2>Rocket 1</h2><p>This is a rocket.</p><button class=\\"rocket-btn reserve-btn\\" type=\\"button\\">Reserve Rocket</button></span></li>"');
  });
});
