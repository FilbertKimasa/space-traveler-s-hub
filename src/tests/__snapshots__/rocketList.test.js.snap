// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RocketItem component renders rocket item correctly 1`] = `
<div>
  <li
    class="rocket-item"
  >
    <img
      alt="rocket"
      class="rocket-img"
      src="rocket-image.jpg"
    />
    <span
      class="display-flex rocket-info"
    >
      <h2>
        Rocket 1
      </h2>
      <p>
        This is a rocket.
      </p>
      <button
        class="reserve-btn"
        type="button"
      >
        Reserve Rocket
      </button>
    </span>
  </li>
</div>
`;
